src/main/java/io/kestra/plugin/microsoft365/outlook/MailReceived.java

```diff
@@ -0,0 +1,500 @@
+                """
+        ),
+        @Example(
+            title = "Wait for emails with attachments and process them",
+            full = true,
+            code = """
+                id: outlook_with_attachments
+                namespace: company.team
+
+                tasks:
+                  - id: each
+                    type: io.kestra.plugin.core.flow.EachSequential
+                    values: "{{ trigger.messages }}"
+                    tasks:
+                      - id: process
+                        type: io.kestra.plugin.core.debug.Return
+                        format: "Processing {{ taskrun.value.subject }} with {{ taskrun.value.attachments | length }} attachments"
+                      - id: each_attachment
+                        type: io.kestra.plugin.core.flow.EachSequential
+                        values: "{{ taskrun.value.attachments }}"
+                        tasks:
+                          - id: log_attachment
+                            type: io.kestra.plugin.core.log.Log
+                            message: "Attachment: {{ taskrun.value.name }} stored at {{ taskrun.value.uri }}"
+
+                triggers:
+                  - id: watch
+                    type: io.kestra.plugin.microsoft365.outlook.MailReceived
+                    interval: PT2M
+                    auth:
+                      tenantId: "{{ secret('AZURE_TENANT_ID') }}"
+                      clientId: "{{ secret('AZURE_CLIENT_ID') }}"
+                      clientSecret: "{{ secret('AZURE_CLIENT_SECRET') }}"
+                      userPrincipalName: "invoices@company.com"
+                    folderId: "inbox"
+                    filter: "hasAttachments eq true"
+                    includeAttachments: true
+                """
+        ),
+        @Example(
+            title = "Wait for emails from specific sender",
+            full = true,
+            code = """
+                id: outlook_filtered_sender
+                namespace: company.team
+
+                tasks:
+                  - id: process
+                    type: io.kestra.plugin.core.log.Log
+                    message: "Received {{ trigger.count }} email(s) from important sender"
+
+                triggers:
+                  - id: watch
+                    type: io.kestra.plugin.microsoft365.outlook.MailReceived
+                    interval: PT1M
+                    auth:
+                      tenantId: "{{ secret('AZURE_TENANT_ID') }}"
+                      clientId: "{{ secret('AZURE_CLIENT_ID') }}"
+                      clientSecret: "{{ secret('AZURE_CLIENT_SECRET') }}"
+                      userPrincipalName: "user@example.com"
+                    filter: "from/emailAddress/address eq 'boss@company.com'"
+                    stateTtl: P30D
+                """
+        )
+    }
+)
+public class MailReceived extends AbstractTrigger implements PollingTriggerInterface, TriggerOutput<MailReceived.Output>, StatefulTriggerInterface {
```

Comment:
`MailReceivedTrigger` also in the examples:

```suggestion
public class MailReceivedTrigger extends AbstractTrigger implements PollingTriggerInterface, TriggerOutput<MailReceived.Output>, StatefulTriggerInterface {
```